bin/peas_shell

```diff
@@ -164,6 +165,13 @@ class PanSensorShell(cmd.Cmd):
         self.weather = AAGCloudSensor(serial_address=port, store_result=True)
         self.do_enable_sensor('weather')
 
+    def do_clear_buffer(self, sensor , sensor_name):
+        if hasattr(self, sensor) and sensor in self.active_sensors:
+            port = self.config[sensor][sensor_name]['serial_port']
+            ser=serial.Serial(port)
```

Comment:
The `self.active_sensors` holds actual sensor objects, either a `ArduinoSerialMonitor` or `AAGCloudSensor`. Those object already have a `SerialData` connection, so we don't want to create a new serial device connection here but instead use the existing ones.

Also, we don't seem to have any issues with the AAGCloudSensor as it is handling its own `read` and `write` methods so clears them appropriately.

The more I think about it the more we might just want to try to manage the buffer better via our read/writes in `pocs.utils.rs232.SerialData` and not expose this command. Specifically, I think we just need to reset the input buffer every time we are trying to issue a write command. @jamessynge thoughts?

Reply:
Actually i did not thought about the pre-existing serial connection. Regardless of the issue #809 yeah the fix that you recently mentioned won't need an extra `do_clear_buffer` method instead it will only require the `self.ser.reset_input_buffer()` , also the `reset_input_buffer()` and `reset_output_buffer()` methods in the same rs232.py file won't be needed.